### PR TITLE
Focus editor

### PR DESCRIPTION
**Pull request by knatten**
This implements the FocusEditor/FocusDesigner system.

A tip to the reviewer: I think it is easiest to review https://github.com/fusetools/Fuse.SublimePlugin/commit/e9c3f16984d1f8ccd84035b794bd961086be154a in isolation, and the changes as one.

### CODE REVIEW

File: fuse.py [hunk 1]
```diff
@@ -463,6 +481,37 @@ def run(self):
...
+		if response == None:
+			log().info("No response for 'focus designer'")
+			return
+
+		if response.status != "Success":
+			log().error("Error in 'focus designer': '%s'", response.status)
+			return
+
+	def is_enabled(self):
+		return os.path.splitext(self.view.file_name())[1].lower() == ".ux"
```
Comment (Tapped) [inline fuse.py]: We have a method called `getExtension` in `fuse_util.py` that can be used instead, and is more correct.

File: fuse.py [hunk 2]
```diff
@@ -436,6 +448,12 @@ def is_enabled(self, type, paths = []):
 def contains_unoproj(path):
 	return os.path.isdir(path) and len([f for f in os.listdir(path) if os.path.isfile(os.path.join(path, f)) and f.lower().endswith(".unoproj")])
 
+def save_open_views():
+	if getSetting("fuse_auto_save_enabled"):
+		for view in sublime.active_window().views():
```
Comment (Tapped) [inline fuse.py]: Don't save all views, just the active view

File: msg_parser.py [hunk 3]
```diff
@@ -99,7 +116,9 @@ def parse(self, message):
 					self.requestsPending[resId]["callback"](response)
 		elif messageType == "Event":
 			return Event(messageParsed["Name"], messageParsed["Data"])
+		elif messageType == "Request":
+			return Request(messageParsed["Name"], messageParsed["Id"], messageParsed["Arguments"])
 		else:
-			print("Fuse: Message type unknown.")
+			logging.getLogger(__name__).info("Fuse: Message type '" + messageType + "' unknown.")
```
Comment (Tapped) [inline msg_parser.py]: Replace `.info()` with `.warning()`

**Comment by knatten** [comment]
Thanks a lot for the review @Tapped, I've now followed up on all the comments.

File: fuse_util.py [hunk 4]
```diff
@@ -30,7 +36,7 @@ def getExtension(path):
 	if ext is None:
 		return ""
 	else:
-		return ext[0]
+		return ext[1].strip(".")
```
Comment (Tapped) [inline fuse_util.py]: I think this is dangerous, since the path may not have more than one part

Comment (knatten) [inline fuse_util.py]: `ext` is not a path, but a return value from [splitext](https://docs.python.org/3/library/os.path.html#os.path.splitext), which always returns two components. Anyway, I added some unit tests now that demonstrates that this is safe.

**Pull request merged**